src/avram/associations/has_many.cr

```diff
@@ -19,12 +19,16 @@ module Avram::Associations::HasMany
       relationship_type: :has_many
 
     {% model = type_declaration.type %}
+    {% singular_through = nil %}
+    {% if !through.is_a?(NilLiteral) %}
+      {% singular_through = run("../../run_macros/singularize_word.cr", through) %}
```

Comment:
I am curious why do we need to use the `run_macro`?

Reply:
Because of the way Avram works, we're not able to look at the associations of other models, so there's no way to know if the association we are referencing is a `has_many` (plural) or a `belongs_to` (singular). So a library could have many authors through books, but from library how do we know if books belongs to an author, which would be singular, or have many authors plural? So I added the singularize word macro to turn `authors` into `author` so that we can check which name for the association that book has.

I do not like this solution AT ALL, but with the current limitations of Avram, I don't see a better way of jumping associations.